test/LGTV.test.ts

```diff
@@ -0,0 +1,159 @@
+import { createSocket } from 'dgram';
+import { Server } from 'net';
+import { promisify } from 'util';
+import { afterEach, beforeEach, describe, expect, it } from 'vitest';
+
+import { LGEncryption } from '../src/classes/LGEncryption.js';
+import { LGTV } from '../src/classes/LGTV.js';
+import { DefaultSettings } from '../src/constants/DefaultSettings.js';
+
+const CRYPT_KEY = 'M9N0AZ62';
+const MAC = 'DA:0A:0F:E1:60:CB';
+
+function stripPadding(padded: string) {
+  return padded.substring(0, padded.indexOf('\r'));
+}
+
+describe('LGTV', () => {
+  it('constructs with valid parameters', () => {
+    const tv = new LGTV('127.0.0.1', MAC, CRYPT_KEY);
+    expect(tv).toBeTruthy();
+  });
+
+  it('constructs with IPv6', () => {
+    const tv = new LGTV('::1', MAC, CRYPT_KEY);
+    expect(tv).toBeTruthy();
+  });
+
+  it('constructs without a MAC address', () => {
+    const tv = new LGTV('127.0.0.1', null, CRYPT_KEY);
+    expect(tv).toBeTruthy();
+  });
+});
+
+describe.each([
+  { ipProto: 'IPv4', address: '127.0.0.1' },
+  { ipProto: 'IPv6', address: '::1' },
+])('streaming commands $ipProto', ({ address }) => {
+  let mockCrypt;
+  let mockServer;
+  let mockSocket;
+  let testSettings;
+  let testTV;
+  let doneMocking;
+  let finishMock;
+
+  beforeEach(() => {
+    mockCrypt = new LGEncryption(CRYPT_KEY);
+    mockServer = new Server((socket) => (mockSocket = socket));
+    mockServer.listen();
+    const port = mockServer.address().port;
+    testSettings = { ...DefaultSettings, networkPort: port };
+    testTV = new LGTV(address, MAC, CRYPT_KEY, testSettings);
+    doneMocking = new Promise((resolve) => (finishMock = resolve));
+  });
+
+  afterEach(async () => {
+    mockSocket?.destroy();
+    mockSocket = undefined;
+    await promisify(mockServer.close).bind(mockServer)();
+  });
+
+  it('connects', async () => {
+    let connected = false;
+    mockServer.on('connection', () => {
+      connected = true;
+      finishMock();
+    });
+
+    await testTV.connect();
+    await doneMocking;
```

Comment:
I wonder if this would be equivalent and remove the need for `doneMocking` and `finishMock`, which are a bit confusing until you read the tests.

```suggestion
    const connection = new Promise((resolve) => {
      mockServer.on('connection', () => {
        connected = true;
        resolve();
      });
    });

    await testTV.connect();
    await connection;
```

Reply:
Yep. Works.